### DISCUSSION TITLE
Localized Zoom

### DISCUSSION THREAD

**Discussion by samcarey** [emilk/egui#261] [Q&A]
I'm trying to plot a graph where nodes are rectangles full of text. I have a top menu bar and sidebars, with the graph in the central panel. I was thinking a nice feature would be to be able to zoom in and out on the graph using the mouse wheel, while the rest of the UI stays the same scale.

My first thought would be to adjust the pixels per point while drawing the central panel and then adjust it back for the other panels, but that seems to only be possible at a global scale since it only applies to the next frame. 

The only other thing I can think of is to individually scale everything in the graph (rectangles, edge lines, thicknesses, fonts) according to some factor. Does that seem like the best approach?

**Comment by emilk** [comment]
Yes, currently you will have to do all your zooming (and translation, rotation, etc) manually as egui doesn't have any transform functionality built-in (everything is in the global screen coordinate system).
